vm/interpreter/src/gas_tracker/mod.rs

```diff
@@ -25,16 +25,27 @@ impl GasTracker {
     /// enough gas remaining for charge.
     pub fn charge_gas(&mut self, charge: GasCharge) -> Result<(), ActorError> {
         let to_use = charge.total();
-        let used = self.gas_used + to_use;
-        if used > self.gas_available {
-            self.gas_used = self.gas_available;
-            Err(actor_error!(SysErrOutOfGas;
-                    "not enough gas (used={}) (available={})",
-               used, self.gas_available
-            ))
-        } else {
-            self.gas_used += to_use;
-            Ok(())
+        let used_or = self.gas_used.checked_add(to_use);
```

Comment:
I would not complicate the Rust code here and distinguish the two cases (`None`/`Some`).
Just use the equivalent algorithm than what Forest does (it's really overflow safe).

So the code would be something like:
```
if self.gas_used > self.gas_available-to_use {
    ...
} else {
    self.gas_used += to_use;
    Ok(())
}
```

Reply:
it just wasn't immediately evident to me why it was overflow safe (I had to think for a second, and could have seen tired-me or someone else changing it back to the overflowable form)

but now I realize this problem is solvable with the highly advanced technology of "commenting your code"

will change